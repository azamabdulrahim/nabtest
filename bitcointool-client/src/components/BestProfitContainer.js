import React, { Component } from "react";
import { getBestProfits } from "../actions/historyActions";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import BestProfitTable from "./BestProfitTable";

export class BestProfitContainer extends Component {
  componentDidMount() {
    const { refresh, date, currency } = this.props;

    if (refresh && date && currency) {
      this.props.getBestProfits(date, currency);
    }
  }

  render() {
    //const { bestProfits } = this.props.history;
    let displayResult;
    let bestProfits;
    if (this.props.history) {
      bestProfits = this.props.history.bestProfits;
    }

    if (bestProfits && bestProfits.length > 0) {
      displayResult = (
        <div>
          {bestProfits.map(data => (
            <BestProfitTable key={data.currency} data={data} />
          ))}
        </div>
      );
    } else {
      displayResult = <p>No data</p>;
    }

    return (
      <div>
        <p />
        <h2>Best Profits</h2>
        {displayResult}
      </div>
    );
  }
}

BestProfitContainer.propTypes = {
  history: PropTypes.object,
  getBestProfits: PropTypes.func,
  errors: PropTypes.object
};

const mapStateToProps = state => ({
  history: state.history,
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { getBestProfits }
)(BestProfitContainer);
