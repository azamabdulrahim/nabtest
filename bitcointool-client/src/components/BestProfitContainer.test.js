import React from "react";
import { shallow, mount } from "enzyme";
import configureStore from "redux-mock-store";
import { BestProfitContainer } from "./BestProfitContainer";

describe("BestProfitContainer", () => {
  // create any initial state needed
  const initialState = {};
  // here it is possible to pass in any middleware if needed into //configureStore
  const mockStore = configureStore();
  let wrapper;

  const store = mockStore(initialState);
  const mockGetBestProfits = jest.fn();

  it("renders correctly when refresh is false", () => {
    wrapper = shallow(
      <BestProfitContainer store={store} getBestProfits={mockGetBestProfits} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div").length).toEqual(1);
    expect(wrapper.containsMatchingElement(<p>No data</p>)).toBeTruthy();
    expect(mockGetBestProfits.mock.calls.length).toBe(0);
  });

  it("renders correctly when refresh is true but date and currency not supplied", () => {
    const refresh = true;
    wrapper = shallow(
      <BestProfitContainer
        store={store}
        refresh={refresh}
        getBestProfits={mockGetBestProfits}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div").length).toEqual(1);
    expect(wrapper.containsMatchingElement(<p>No data</p>)).toBeTruthy();
    expect(mockGetBestProfits.mock.calls.length).toBe(0);
  });

  it("renders correctly when refresh is true but date not supplied", () => {
    const refresh = true;
    const currency = "BTC";
    wrapper = shallow(
      <BestProfitContainer
        store={store}
        refresh={refresh}
        currency={currency}
        getBestProfits={mockGetBestProfits}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div").length).toEqual(1);
    expect(wrapper.containsMatchingElement(<p>No data</p>)).toBeTruthy();
    expect(mockGetBestProfits.mock.calls.length).toBe(0);
  });

  it("renders correctly when refresh is true but currency not supplied", () => {
    const refresh = true;
    const date = "20180507";
    wrapper = shallow(
      <BestProfitContainer
        store={store}
        refresh={refresh}
        date={date}
        getBestProfits={mockGetBestProfits}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div").length).toEqual(1);
    expect(wrapper.containsMatchingElement(<p>No data</p>)).toBeTruthy();
    expect(mockGetBestProfits.mock.calls.length).toBe(1);
  });

  it("calls getBestProfits function and renders child components when refresh is true", () => {
    const mockGetBestProfits = jest.fn();
    const refresh = true;
    const currency = "BTC";
    const date = "20180507";
    const history = {
      bestProfits: [
        {
          date: "07/05/2018",
          currency: "BTC",
          buy: {
            time: "09:30 AM",
            price: "2.30"
          },
          sell: {
            time: "04:30 PM",
            price: "2.58"
          },
          profit: "0.28"
        },
        {
          date: "07/05/2018",
          currency: "ETC",
          buy: {
            time: "08:30 AM",
            price: "1.30"
          },
          sell: {
            time: "12:30 PM",
            price: "1.50"
          },
          profit: "0.20"
        }
      ]
    };

    wrapper = shallow(
      <BestProfitContainer
        store={store}
        refresh={refresh}
        date={date}
        currency={currency}
        getBestProfits={mockGetBestProfits}
        history={history}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(mockGetBestProfits).toHaveBeenCalled();
    expect(wrapper.find("BestProfitTable").length).toEqual(2);
  });
});
