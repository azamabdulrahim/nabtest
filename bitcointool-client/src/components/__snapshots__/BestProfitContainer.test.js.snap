// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestProfitContainer calls getBestProfits function and renders child components when refresh is true 1`] = `
<div>
  <p />
  <h2>
    Best Profits
  </h2>
  <div>
    <BestProfitTable
      data={
        Object {
          "buy": Object {
            "price": "2.30",
            "time": "09:30 AM",
          },
          "currency": "BTC",
          "date": "07/05/2018",
          "profit": "0.28",
          "sell": Object {
            "price": "2.58",
            "time": "04:30 PM",
          },
        }
      }
      key="BTC"
    />
    <BestProfitTable
      data={
        Object {
          "buy": Object {
            "price": "1.30",
            "time": "08:30 AM",
          },
          "currency": "ETC",
          "date": "07/05/2018",
          "profit": "0.20",
          "sell": Object {
            "price": "1.50",
            "time": "12:30 PM",
          },
        }
      }
      key="ETC"
    />
  </div>
</div>
`;

exports[`BestProfitContainer renders correctly when refresh is false 1`] = `
<div>
  <p />
  <h2>
    Best Profits
  </h2>
  <p>
    No data
  </p>
</div>
`;

exports[`BestProfitContainer renders correctly when refresh is true but currency not supplied 1`] = `
<div>
  <p />
  <h2>
    Best Profits
  </h2>
  <p>
    No data
  </p>
</div>
`;

exports[`BestProfitContainer renders correctly when refresh is true but date and currency not supplied 1`] = `
<div>
  <p />
  <h2>
    Best Profits
  </h2>
  <p>
    No data
  </p>
</div>
`;

exports[`BestProfitContainer renders correctly when refresh is true but date not supplied 1`] = `
<div>
  <p />
  <h2>
    Best Profits
  </h2>
  <p>
    No data
  </p>
</div>
`;
